### DISCUSSION TITLE
Why does gd command on Linux show a git error?

### DISCUSSION THREAD

**Discussion by ardawan** [quaadgras/graphics.gd#7] [Q&A] [answered]
1.  installed Godot 4.2.1 via `snap install godot-4` this is not Mono version.
2. created a project `/my/dir/`
3. run `go install grow.graphics/gd/cmd/gd@master`
4. created an empty main function in `/my/dir/main.go`
5. run `gd`, `gd run`, `gd test` all displays is:

```
$ gd
warning: Not a git repository. Use --no-index to compare two paths outside a working tree
usage: git diff --no-index [<options>] <path> <path>

Diff output format options
    -p, --patch           generate patch
    -s, --no-patch        suppress diff output
    -u                    generate patch
    -U, --unified[=<n>]   generate diffs with <n> lines context
    -W, --function-context
...
```

**Comment by Splizard** [comment, marked as answer]
Looks like you've setup `gd`  as an alias for `git diff`, so it's not running the `grow.graphics/gd/cmd/gd`.

https://jonsuh.com/blog/git-command-line-shortcuts/

Check that `$(go env GOPATH)/bin` is in your `$PATH` and consider removing the alias if you are not `git diff`ing regularly.

**Reply by ardawan** [reply]
I'm using zsh and the path to those aliases wasn't in `~/.bash_profile` , `~/.bashrc` or `~/.zshrc`. 
instead, I found it in `~/.oh-my-zsh/plugins/git/git.plugin.zsh`

now I can use the command but need to find out about those ERRORs:

```
$ gd run
ERROR: Can't resolve symbol loadExtension. Error: /home/x//go_game/graphics/library.so: undefined symbol: loadExtension.
   at: get_dynamic_library_symbol_handle (drivers/unix/os_unix.cpp:684)
ERROR: GDExtension entry point 'loadExtension' not found in library /home/x/go_game/graphics/library.so
   at: open_library (core/extension/gdextension.cpp:736)
ERROR: Failed loading resource: res://library.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:274)
ERROR: Error loading extension: res://library.gdextension
   at: load_extensions (core/extension/gdextension_manager.cpp:234)
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
Vulkan API 1.3.238 - Forward+ - Using Vulkan Device #0: AMD - AMD Unknown (RADV RENOIR)
```

**Reply by Splizard** [reply]
Ah, yes make sure you import the `grow.graphics/gd/gdextension` package in your file. Ie. Call `Link`.  Also start the editor once so that it can initialise. I'll look into a way to do this automatically when you start a new project.

**Reply by tizu69** [reply]
oh-my-zsh has a default git plugin which does this.

I think naming the cli `gd` is a bit confusing tbh. I'd propose an optional rename (probably by just providing a sister package that calls the normal gd cli right through go, so it does the same?) to something like `gogd`, `gdgo`, or something similar. A two-letter cli like this is... not something I'd want to mess with myself tbh, but once it works it works I guess.

**Reply by Splizard** [reply]
you can always use `go tool` if you prefer
